libcodechecker/log/log_parser.py

```diff
@@ -0,0 +1,748 @@
+    # information depends on other data like language or target architecture.
+    compiler_includes = {}
+    compiler_target = {}
+    compiler_standard = {}
+
+    def __get_compiler_err(self, cmd):
+        """
+        Returns the stderr of a compiler invocation as string.
+        """
+        try:
+            LOG.debug("Retrieving default includes via '" + cmd + "'")
+            proc = subprocess.Popen(shlex.split(cmd),
+                                    stdin=subprocess.PIPE,
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+
+            _, err = proc.communicate("")
+            return err
+        except OSError as oerr:
+            LOG.error("Error during process execution: " + cmd + '\n' +
+                      oerr.strerror + "\n")
+
+    def __parse_compiler_includes(self, lines):
+        """
+        Parse the compiler include paths from a string
+        """
+        start_mark = "#include <...> search starts here:"
+        end_mark = "End of search list."
+
+        include_paths = []
+
+        do_append = False
+        for line in lines.splitlines():
+            if line.startswith(end_mark):
+                break
+            if do_append:
+                # On OSX there are framework includes,
+                # where we need to strip the "(framework directory)" string.
+                # For instance:
+                # /System/Library/Frameworks (framework directory)
+                fpos = line.find("(framework directory)")
+                if fpos == -1:
+                    include_paths.append(line)
+                else:
+                    include_paths.append(line[:fpos - 1])
+
+            if line.startswith(start_mark):
+                do_append = True
+
+        return include_paths
+
+    def __filter_compiler_includes(self, include_dirs):
+        """
+        Filter the list of compiler includes.
+        We want to elide GCC's include-fixed and instrinsic directory.
+        See docs/gcc_incompatibilities.md
+        """
+
+        def contains_intrinsic_headers(include_dir):
+            """
+            Returns True if the given directory contains at least one intrinsic
+            header.
+            """
+            if not os.path.exists(include_dir):
+                return False
+            for f in os.listdir(include_dir):
+                if f.endswith("intrin.h"):
+                    return True
+            return False
+
+        result = []
+        for include_dir in include_dirs:
+            # Skip GCC's fixinclude dir
+            if os.path.basename(
+                    os.path.normpath(include_dir)) == "include-fixed":
+                continue
+            if contains_intrinsic_headers(include_dir):
+                continue
+            result.append(include_dir)
+        return result
+
+    def get_compiler_includes(self, action):
```

Comment:
compiler, lang and analyzer options should be enough as an argument we do not use any other information from the buildaction.

Reply:
I think this is for getting the default include paths from the compiler as it might be different from clang's default include paths. These include paths might not appear in the compilation command, we might still want to add them.